File: Non-constructible change/sol.py
# O(nlogn) time and O(1) space

def nonConstructibleChange(coins):
    # Write your code here.
    coins.sort()
    minConstructableChange = 0
    for coin in coins:
        if coin > minConstructableChange + 1:
            break
        minConstructableChange+=coin
    return minConstructableChange+1

File: Non-constructible change/sol.js
// O(nlogn) time and O(1) space

function nonConstructibleChange(coins) {
  // Write your code here.
  coins.sort((a, b) => a - b);
  let minConstructibleChange = 0;
  for (let coin of coins) {
    if (coin > minConstructibleChange + 1) break;
    minConstructibleChange += coin;
  }
  return minConstructibleChange + 1;
}
